visualization/SensorVisualization/RoboSailSerialParser.pde

```diff
@@ -0,0 +1,87 @@
+/*
+ * RoboSailSerialParser
+ *
+ * (c) 2016 Brian Dolan, RoboSail
+ *
+ * (See description for base class, SerialParser.)  The RobSailSerialParser implementation works for 
+ * the "real" output from one of the standard RoboSail Arduino programs used in the RoboSail program.
+ */
+import processing.serial.*;
+
+public class RoboSailSerialParser extends SerialParser
+{
+  RoboSailSerialParser(PApplet parent, String comPort)
+  {
+    super(parent, comPort);
+  }
+  
+  // parse the serial output, in the format of the Arduino program's output, which is
+  // int the format:
+  // 
+  // [joystick x value (0-1023)] [joystick y value (0-1023)] [B1][B2][B3][B4][B5]
+  //
+  // where B1-B1 are 0 or 1 values representing the state of the 5 buttons 
+  // 
+  // this function must be "hooked up" to the global serialEvent() function by the main 
+  // application, e.g. 
+  // void serialEvent(Serial sPort) { serialParser.serialEvent(sPort); }
```

Comment:
Can you add a bit more on the expected serial format with an example?

As a side note, for systems like these the format is usually just a CSV like `1,34,543.5,3454,34.66,343`, where the full state is output every cycle. It's efficient on the wire and allows the receiver to be stateless (ie, it doesn't have to remember things between sentences, which makes error handling easier).

In other cases it's like NMEA, where you only get partial state for every sentence. UGH.